src/org/traccar/FilterHandler.java

```diff
@@ -127,7 +133,7 @@ private boolean filterMaxSpeed(Position position, Position last) {
         if (filterMaxSpeed != 0 && last != null) {
             double distance = position.getDouble(Position.KEY_DISTANCE);
             long time = position.getFixTime().getTime() - last.getFixTime().getTime();
-            return UnitsConverter.knotsFromMps(distance / (time / 1000)) > filterMaxSpeed;
+            return UnitsConverter.knotsFromMps(distance / (time / 1000.0)) > filterMaxSpeed;
```

Comment:
Why do we need decimal point here?

Reply:
Here is FindBug complain http://findbugs.sourceforge.net/bugDescriptions.html#ICAST_IDIV_CAST_TO_DOUBLE
And I agree with it, we are loosing some precision.

Reply:
In most cases devices report only seconds, so we don't lose anything. Also, usually max speed is very high value, so 0.1% doesn't make much difference.

Anyway, I just don't like how it looks. Maybe we can change variable from `long` to `double`?